pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1990,10 +1974,10 @@ SEC("kprobe/security_file_ioctl")
 int BPF_KPROBE(trace_security_file_ioctl)
 {
     program_data_t p = {};
-    if (!init_program_data(&p, ctx))
+    if (!init_program_data(&p, ctx, BPF_ATTACH))
         return 0;
 
-    if (!should_trace(&p))
+    if (!evaluate_scope_filters(&p))
```

Comment:
Maybe, however, I prefer to keep the programs structured since in the future we will probably be splitting them to different programs that will tail call one another (scope filters->argument collection->argument filtering->submit the event)